biz.aQute.bndlib/src/aQute/bnd/build/model/BndEditModel.java

```diff
@@ -677,12 +685,86 @@ public void setIncludeSources(boolean includeSources) {
 	}
 
 	public List<String> getPrivatePackages() {
-		return doGetObject(aQute.bnd.osgi.Constants.PRIVATE_PACKAGE, listConverter);
+		List<String> privatePackagesEntries1 = getEntries(Constants.PRIVATEPACKAGE, listConverter);
+		List<String> privatePackagesEntries2 = getEntries(Constants.PRIVATE_PACKAGE, listConverter);
+
+		Set<String> privatePackages = Stream.concat(privatePackagesEntries1.stream(), privatePackagesEntries2.stream())
+			.collect(toSet());
+		return new ArrayList<>(privatePackages);
+	}
+
+	public void setPrivatePackages(List<String> newPackages) {
+		List<String> privatePackagesEntries1 = getEntries(Constants.PRIVATEPACKAGE, listConverter);
+		List<String> privatePackagesEntries2 = getEntries(Constants.PRIVATE_PACKAGE, listConverter);
+		
+		Set<String> privatePackages = Stream.concat(privatePackagesEntries1.stream(), privatePackagesEntries2.stream())
+			.collect(toSet());
+		
+		List<String> addedEntries = disjunction(newPackages, privatePackages);
+		List<String> removedEntries = disjunction(privatePackages, newPackages);
+
+		privatePackagesEntries1.removeAll(removedEntries);
+		if (privatePackagesEntries1.isEmpty()) {
+			removeEntries(Constants.PRIVATEPACKAGE);
+		} else {
+			setEntries(privatePackagesEntries1, Constants.PRIVATEPACKAGE);
+		}
+
+		privatePackagesEntries2.removeAll(removedEntries);
+		if (privatePackagesEntries2.isEmpty()) {
+			removeEntries(Constants.PRIVATE_PACKAGE);
+		} else {
+			setEntries(privatePackagesEntries2, Constants.PRIVATE_PACKAGE);
+		}
+		
+		if (hasPrivatePackageInstruction()) {
+			privatePackagesEntries1.addAll(addedEntries);
+			setEntries(privatePackagesEntries1, Constants.PRIVATEPACKAGE);
+		} else {
+			privatePackagesEntries2.addAll(addedEntries);
+			setEntries(privatePackagesEntries2, Constants.PRIVATE_PACKAGE);
+		}
 	}
 
-	public void setPrivatePackages(List<? extends String> packages) {
-		List<String> oldPackages = getPrivatePackages();
-		doSetObject(aQute.bnd.osgi.Constants.PRIVATE_PACKAGE, oldPackages, packages, stringListFormatter);
+	public void setEntries(List<? extends String> packages, String key) {
```

Comment:
We should make all the new methods you have added `private` instead of `public`. Unless there is a real reason for the new methods to be API, lets not make them API.

Reply:
Baselining does tend you to make a lot more conservative regarding public! :-)

Reply:
It also preserves your future freedom to not make things public unless necessary.

Reply:
Agreed and fixed 👍